opensearch-project/opensearch-dashboards-functional-test
# Add precommit hook to run lint command

### Description

* Use Husky precommit hook to automatically run `yarn lint` when committing code changes

### Issues Resolved

https://github.com/opensearch-project/opensearch-dashboards-functional-test/issues/198

### Testing

* Test 1: there are errors when running precommit hook
  * Add prettier error in one of files
  * Commit the code change
  * See the error message
```
ubuntu@ip-172-31-55-176:~/github/opensearch-dashboards-functional-test$ git commit -s -m "test"
yarn run v1.22.10
$ eslint . --ext .js

/home/ubuntu/github/opensearch-dashboards-functional-test/cypress/integration/plugins/alerting-dashboards-plugin/bucket_level_monitor_spec.js
  9:8  error  Replace `"../../../utils/plugins/alerting-dashboards-plugin/constants"` with `'../../../utils/plugins/alerting-dashboards-plugin/constants'`  prettier/prettier

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
husky - pre-commit hook exited with code 1 (error)
```
* Test 2: no errors when running precommit hook
  * Fix the error by running `yarn lint --fix`
  * Commit the code change
  * See precommit hook passed
```
ubuntu@ip-172-31-55-176:~/github/opensearch-dashboards-functional-test$ git commit -s -m "add precommit hook to run lint command"
yarn run v1.22.10
$ eslint . --ext .js
Done in 2.46s.
[conjwang/precommit 5d02ec0] add precommit hook to run lint command
 4 files changed, 12 insertions(+)
 create mode 100644 .husky/.gitignore
 create mode 100755 .husky/pre-commit
```

### Check List


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
